operator/duties/committee.go

```diff
@@ -97,39 +97,31 @@ func (h *CommitteeHandler) buildCommitteeDuties(attDuties []*eth2apiv1.AttesterD
 	}
 
 	resultCommitteeMap := make(committeeDutiesMap)
-
 	for _, duty := range attDuties {
-		committee, ok := validatorCommittees[duty.ValidatorIndex]
-		if !ok {
-			h.logger.Error("failed to find committee for validator", zap.Uint64("validator_index", uint64(duty.ValidatorIndex)))
-			continue
-		}
-
 		if h.shouldExecuteAtt(duty, epoch) {
-			addToCommitteeMap(resultCommitteeMap, committee, h.toSpecAttDuty(duty, spectypes.BNRoleAttester))
+			h.addToCommitteeMap(resultCommitteeMap, validatorCommittees, h.toSpecAttDuty(duty, spectypes.BNRoleAttester))
 		}
 	}
-
 	for _, duty := range syncDuties {
-		committee, ok := validatorCommittees[duty.ValidatorIndex]
-		if !ok {
-			h.logger.Error("failed to find committee for validator", zap.Uint64("validator_index", uint64(duty.ValidatorIndex)))
-			continue
-		}
-
 		if h.shouldExecuteSync(duty, slot, epoch) {
-			addToCommitteeMap(resultCommitteeMap, committee, h.toSpecSyncDuty(duty, slot, spectypes.BNRoleSyncCommittee))
+			h.addToCommitteeMap(resultCommitteeMap, validatorCommittees, h.toSpecSyncDuty(duty, slot, spectypes.BNRoleSyncCommittee))
 		}
 	}
 
 	return resultCommitteeMap
 }
 
-func addToCommitteeMap(
+func (h *CommitteeHandler) addToCommitteeMap(
 	committeeDutyMap committeeDutiesMap,
-	committee committeeDuty,
+	validatorCommittees map[phase0.ValidatorIndex]committeeDuty,
 	specDuty *spectypes.ValidatorDuty,
 ) {
+	committee, ok := validatorCommittees[specDuty.ValidatorIndex]
+	if !ok {
+		h.logger.Error("failed to find committee for validator", zap.Uint64("validator_index", uint64(specDuty.ValidatorIndex)))
+		return
+	}
```

Comment:
It's a bit confusing this function can fail, it should've probably been in `shouldExecute..` , but it's also good to keep the should functions around ethereum eligibility and not internal issues.